sklearn/multioutput.py

```diff
@@ -162,10 +162,11 @@ def partial_fit(self, X, y, classes=None, sample_weight=None, **partial_fit_para
             )
 
         if _routing_enabled():
+            if sample_weight is not None:
+                partial_fit_params["sample_weight"] = sample_weight
```

Comment:
We modify the object provided by the user. I don't think we can do that.

Reply:
We should create a new dictionary instead.

Reply:
the user never passes a dictionary here, it's the kwargs.

```python
In [1]: def f(**kwargs):
   ...:     kwargs[10] = 0
   ...: 

In [2]: a = {'a':2}

In [3]: f(a)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[3], line 1
----> 1 f(a)

TypeError: f() takes 0 positional arguments but 1 was given

In [4]: f(**a)

In [5]: a
Out[5]: {'a': 2}
```

Reply:
Indeed. I was tired that day :)